### PR TITLE
added maxFontSizeMultiplier to text ios props

### LINKED ISSUE #33792: [@types/react-native] maxFontSizeMultiplier prop not supported
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors:  @alloy @huhuanming @iRoachie @timwangdev @kamal @nelyousfi @alexdunne @swissmanu @bm-software @tkrotoff @a-tarasyuk @mvdam

If you do not mention the authors the issue will be ignored.

I was trying to use the maxFontSizeMultiplier prop and was receiving an error because it wasn't supported. Would it be possible to add this prop in? 

For reference:
https://facebook.github.io/react-native/docs/text#maxfontsizemultiplier

### PR DESCRIPTION
**Pull request by saranshkataria**
Fixes #33792

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -808,6 +809,14 @@ export interface TextPropsIOS {
...
      */
     suppressHighlighting?: boolean;
+
+    /**
+     * Specifies largest possible scale a font can reach when allowFontScaling is enabled. Possible values:
+     * - null/undefined (default): inherit from the parent node or the global default (0)
+     * - 0: no max, ignore parent/global default
+     * - >= 1: sets the maxFontSizeMultiplier of this node to this value
+     */
+    maxFontSizeMultiplier?: number;
```
Comment (alloy) [inline types/react-native/index.d.ts]: This is missing the `null` typing

```suggestion
    maxFontSizeMultiplier?: number | null;
```

File: types/react-native/index.d.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@
 //                 Ryan Nickel <https://github.com/mrnickel>
 //                 Souvik Ghosh <https://github.com/souvik-ghosh>
 //                 Cheng Gibson <https://github.com/nossbigg>
+//                 Saransh Kataria <https://github.com/saranshkataria>
```
Comment (alloy) [inline types/react-native/index.d.ts]: Is it clear to you that this means to sign-up as a maintainer and you'll be pinged on each PR/issue for react-native?

**Review by alloy — CHANGES_REQUESTED** [review]
Thanks! Can you add some test coverage please?

**Comment by timwangdev** [comment]
Hi @saranshkataria, could you also add the `maxFontSizeMultiplier ` prop to `TextInputProps`? Thanks!

File: types/react-native/index.d.ts (hunk 2 above)
Comment (saranshkataria) [reply types/react-native/index.d.ts]: Yes, I know. Will try and help as much as I can :smile:

**Comment by jessamp** [comment]
Shouldn't this be under TextProps instead of just TextPropsIOS? According to the website, it should work for both, unless I'm misunderstanding.

https://facebook.github.io/react-native/docs/text#maxfontsizemultiplier

**Comment by saranshkataria** [comment]
@timwangdev added

**Comment by saranshkataria** [comment]
@alloy made the suggested changes, not sure about test coverage though. Don't see any tests having TextProps, and not sure what needs to be added. Could you please point me to the correct direction?

**Comment by alloy** [comment]
The test should just be a snippet of code that uses the types. If you can find places to add a snippet using your additions then that’s great, if not you can just add a new snippet of code.

**Comment by saranshkataria** [comment]
I see the tests succeeding locally, but they fail due to trailing whitespace. 

@sandersn am I missing something here? This is npm run test's output for me:

image

**Comment by saranshkataria** [comment]
@alloy added

**Comment by alloy** [comment]
Thanks!

**Review by alloy — APPROVED** [review]

**Pull request merged**